### ISSUE TITLE
Weird glitches with the perch on shoulder function

### ISSUE THREAD

**Issue by KaylebJay** [runsy/petz#30]
I decided to expirement with some of petz's new functions. It seems there is a regression somewhere here because my older version never had this happen with the perching.
Firstly, with the parrot on your shoulder, by walking you get a crazy kickback because of the parrot 'rocking' against the players hitbox. This doesn't happen with the toucan's hitbox though (https://github.com/runsy/petz/pull/29). Also, if you sneak around with a parrot on your shoulder (shift=sneak), and glitch into a block you can get an error like this:
`2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: WARNING[Main]: updateListener: invalid value
2020-05-02 14:41:49: ERROR[Main]: ServerError: AsyncErr: ServerThread::run Lua: Invalid float vector dimension range 'x' (expected -2.14748e+06 < x < 2.14748e+06 got -2.14748e+06).
`

**Comment by KaylebJay** [comment]
Oh, by the way, this is with Termos' new entity physics PR (Minetest 5.3.0-dev-8ef239b4)

**Comment by runsy** [comment]
Please, could you report this to Termos?

**Comment by KaylebJay** [comment]
Definitely.

**Comment by runsy** [comment]
> 
> 
> Definitely.

> 
> 
> Definitely.

Petz does not work with 5.3-dev at this moment, only stable versions.

**Comment by TheTermos** [comment]
I'd have to do some testing,
however it's not a good idea to clip boxes into each other, a parrot sitting on player's shoulder should probably be implemented as an attachment, then collision detection wouldn't apply to it.

**Comment by TheTermos** [comment]
Done some testing - in effect of clicking perch all movement is jittery, even if player is flying.
No collisions occur while flying so this issue can't be related to collision detection.

By the way, on my end the parrot disappears instead of siting on the shoulder.

**Comment by KaylebJay** [comment]
Huh, interesting.

**Comment by runsy** [comment]
Maybe I should open a bug in Minetest.

Kayleb, did it work in 5.2?

**Comment by KaylebJay** [comment]
Yes, the player collides into the parrot's box. It works on 5.2.0.

**Comment by runsy** [comment]
In the new version I set/unset the physical property and seems that solved the issue.

**Comment by KaylebJay** [comment]
If this is solved, https://github.com/runsy/petz/pull/29 should be merged :-)

**Comment by TheTermos** [comment]
By the way I've checked and I think there never was anything to stop localplayer from colliding with own attachments, strange because there's code to prevent entities from doing that.
That might justify opening an issue with MT,

**Comment by runsy** [comment]
> 
> 
> If this is solved, #29 should be merged :-)

The right approach would be:

line 125:
`if pet.type == "parrot" or pet.type == "toucan" then`

line 282:

```
local shoulder_pos
if pet.type == "parrot" then
  shoulder_pos = {x= 0.5, y= -6.25, z=0}
else
  shoulder_pos = {x= 0.5, y= -6.25, z=0}
end
pet.object:set_attach(player, "Arm_Left", shoulder_pos, {x=0, y=0, z=180})
```

but the shoulder_pos values for the toucan should be changed maybe to a better fit of the toucan on the shoulder.

I will do in the next version, of course. :-)

**Comment by KaylebJay** [comment]
Agreed. Thanks.

**Comment by runsy** [comment]
This is an engine issue.
